#! /bin/sh /usr/share/dpatch/dpatch-run
## cht-02_timers.dpatch
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Write timers in such a way that they match channels.conf

@DPATCH@
Index: channels.c
===================================================================
--- vdr-1.6.0.orig/channels.c	(revision 415)
+++ vdr-1.6.0.new/channels.c	(working copy)
@@ -363,6 +363,14 @@
         Channels.SetModified();
         Channels.UnhashChannel(this);
         }
+     if (orig_initialized == 0) {
+        dsyslog("storing original channel id %d-%d-%d-%d", nid, tid, sid, rid);
+        orig_nid = nid;
+        orig_tid = tid;
+        orig_sid = sid;
+        orig_rid = rid;
+		orig_initialized = 1;
+        }
      nid = Nid;
      tid = Tid;
      sid = Sid;
@@ -1000,14 +1008,14 @@
   if (list) {
      for (cHashObject *hobj = list->First(); hobj; hobj = list->Next(hobj)) {
          cChannel *channel = (cChannel *)hobj->Object();
-         if (channel->Sid() == sid && channel->GetChannelID() == ChannelID)
+         if (channel->Sid() == sid && (channel->GetChannelID() == ChannelID || channel->GetOrigChannelID() == ChannelID))
             return channel;
          }
      if (TryWithoutRid) {
         ChannelID.ClrRid();
         for (cHashObject *hobj = list->First(); hobj; hobj = list->Next(hobj)) {
             cChannel *channel = (cChannel *)hobj->Object();
-            if (channel->Sid() == sid && channel->GetChannelID().ClrRid() == ChannelID)
+            if (channel->Sid() == sid && (channel->GetChannelID().ClrRid() == ChannelID || channel->GetOrigChannelID().ClrRid() == ChannelID))
                return channel;
             }
         }
@@ -1015,7 +1023,7 @@
         ChannelID.ClrPolarization();
         for (cHashObject *hobj = list->First(); hobj; hobj = list->Next(hobj)) {
             cChannel *channel = (cChannel *)hobj->Object();
-            if (channel->Sid() == sid && channel->GetChannelID().ClrPolarization() == ChannelID)
+            if (channel->Sid() == sid && (channel->GetChannelID().ClrPolarization() == ChannelID || channel->GetOrigChannelID().ClrPolarization() == ChannelID))
                return channel;
             }
         }
Index: channels.h
===================================================================
--- vdr-1.6.0.orig/channels.h	(revision 415)
+++ vdr-1.6.0.new/channels.h	(working copy)
@@ -132,6 +132,11 @@
   int tid;
   int sid;
   int rid;
+  int orig_nid;
+  int orig_tid;
+  int orig_sid;
+  int orig_rid;
+  int orig_initialized;
   int number;    // Sequence number assigned on load
   bool groupSep;
   char polarization;
@@ -203,6 +208,7 @@
   bool IsSat(void) const { return cSource::IsSat(source); }
   bool IsTerr(void) const { return cSource::IsTerr(source); }
   tChannelID GetChannelID(void) const { return tChannelID(source, nid, (nid || tid) ? tid : Transponder(), sid, rid); }
+  tChannelID GetOrigChannelID(void) const { return tChannelID(source, orig_nid, (orig_nid || orig_tid) ? orig_tid : Transponder(), orig_sid, orig_rid); }
   bool HasTimer(void) const;
   int Modification(int Mask = CHANNELMOD_ALL);
   void CopyTransponderData(const cChannel *Channel);
